nni/experiment/config/kubeflow.py

```diff
@@ -32,37 +32,41 @@ class KubeflowAzureStorageConfig(ConfigBase):
     storage: str = 'azureStorage'
     azure_account: str
     azure_share: str
-    key_vault: str
-    key_vault_secret: str
+    key_vault_name: str
+    key_vault_key: str
 
 
 @dataclass(init=False)
 class KubeflowRoleConfig(ConfigBase):
     replicas: int
     command: str
-    gpu_number: int
+    gpu_number: Optional[int] = 0
     cpu_number: int
     memory_size: str
     docker_image: str = 'msranni/nni:latest'
+    code_directory: str
 
 
 @dataclass(init=False)
 class KubeflowConfig(TrainingServiceConfig):
     platform: str = 'kubeflow'
     operator: str
     api_version: str
-    storage: _KubeflowStorageConfig
-    worker: KubeflowRoleConfig
-    parameter_server: Optional[KubeflowRoleConfig] = None
+    storage: KubeflowStorageConfig
+    worker: Optional[KubeflowRoleConfig] = None
+    ps: Optional[KubeflowRoleConfig] = None
+    master: Optional[KubeflowRoleConfig] = None
+    reuse_mode: Optional[bool] = True #set reuse mode as true for v2 config
 
     def __init__(self, **kwargs):
         kwargs = util.case_insensitive(kwargs)
-        kwargs['storage'] = util.load_config(_KubeflowStorageConfig, kwargs.get('storage'))
+        kwargs['storage'] = util.load_config(KubeflowStorageConfig, kwargs.get('storage'))
         kwargs['worker'] = util.load_config(KubeflowRoleConfig, kwargs.get('worker'))
-        kwargs['parameterserver'] = util.load_config(KubeflowRoleConfig, kwargs.get('parameterserver'))
+        kwargs['ps'] = util.load_config(KubeflowRoleConfig, kwargs.get('ps'))
```

Comment:
`ps` is `parameterserver`? why we don't have `master` before?

Reply:
I once merged ps and master. They are both parameter server. No difference at all, just different names for pytorch and tensorflow.